src/typval.c

```diff
@@ -1758,14 +1751,7 @@ typval_compare_object(
 	return OK;
     }
 
-    for (int i = 0; i < cl1->class_obj_member_count; ++i)
-	if (!tv_equal((typval_T *)(obj1 + 1) + i,
-				 (typval_T *)(obj2 + 1) + i, ic, TRUE))
-	{
-	    *res = !res_match;
-	    return OK;
-	}
-    *res = res_match;
+    *res = object_equal(obj1, obj2, ic, FALSE) ? res_match : !res_match;
```

Comment:
shouldn't we pass TRUE here for recursiveness? Since that is what was done in the lines above that were removed?

Reply:
Nope, that's the source of the problem.
The `recursive` flag is only set from whithin `tv_equal`, that's the assumption that makes the code work. If you check the other cases such as `typval_compare_list` or the dict one you'll see this parameter is false.

Reply:
Ah I see thanks.